src/react/utils.js

```diff
@@ -1153,7 +1153,18 @@ export function hardModifyReactObjectPropertyBinding(
     "hardModifyReactObjectPropertyBinding can only be used on final objects!"
   );
   let binding = object.properties.get(propName);
-  invariant(binding !== undefined);
+  if (binding === undefined) {
+    binding = {
```

Comment:
This looks like we're trying to create an object to match the shape below. Can we instead make peace with it being `undefined`, and remove the invariant below? `Object.assign` should work either way.

Reply:
If it's undefined, then we need to create the binding. That's why we do it here.